src/hotspot/cpu/riscv/gc/shenandoah/shenandoahBarrierSetAssembler_riscv.cpp

```diff
@@ -314,7 +315,7 @@ void ShenandoahBarrierSetAssembler::iu_barrier(MacroAssembler* masm, Register ds
   if (ShenandoahIUBarrier) {
     __ push_call_clobbered_registers();
 
-    satb_write_barrier_pre(masm, noreg, dst, xthread, tmp, true, false);
+    satb_write_barrier_pre(masm, noreg, dst, xthread, tmp, t0, true, false);
```

Comment:
Is `tmp` not `t0` here? I expected to see `t1` in this diff. Maybe I am missing something.

Reply:
PS: Also eyeballed all possible callers of ShenandoahBarrierSetAssembler::satb_write_barrier_pre and I think we are safe here :-)

Reply:
Okay then!